submissions/BlessingOkoro/ecommerce-project/src/Application/Actions/AddProductsAction.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Application\Actions;
+
+use App\Application\Handlers\DbHandlerInterface;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Log\LoggerInterface;
+
+class AddProductsAction extends Action
+{
+    /** @var \PDO $connection */
+    private $connection;
+
+    public function __construct(LoggerInterface $logger, DbHandlerInterface $dbHandlerInterface)
+    {
+        parent::__construct($logger);
+        $this->connection = $dbHandlerInterface->getConnection();
+    }
+    /**
+     * @param Request  $request
```

Comment:
these annotations are not needed here, because:
1. it's already done in the base `Action` class
2. this method does not directly accept those parameters